demos/ota/ota_demo_core_mqtt/ota_demo_core_mqtt.c

```diff
@@ -0,0 +1,1524 @@
+
+    if( returnStatus == EXIT_SUCCESS )
+    {
+        /* Wait till OTA library is stopped, output statistics for currently running
+         * OTA job */
+        while( ( ( state = OTA_GetState() ) != OtaAgentStateStopped ) )
+        {
+            if( mqttSessionEstablished != true )
+            {
+                /* Connect to MQTT broker and create MQTT connection. */
+                returnStatus = establishConnection();
+
+                if( returnStatus == EXIT_SUCCESS )
+                {
+                    mqttSessionEstablished = true;
+
+                    /* Check if OTA process was suspended and resume if required. */
+                    if( state == OtaAgentStateSuspended )
+                    {
+                        /* Resume OTA operations. */
+                        OTA_Resume();
+                    }
+                    else
+                    {
+                        /* Send start event to OTA Agent.*/
+                        eventMsg.eventId = OtaAgentEventStart;
+                        OTA_SignalEvent( &eventMsg );
+                    }
+                }
+            }
+
+            if( mqttSessionEstablished == true )
+            {
+                /* Acquire the mqtt mutex lock. */
+                if( pthread_mutex_lock( &mqttMutex ) == 0 )
+                {
+                    /* Loop to receive packet from transport interface. */
+                    mqttStatus = MQTT_ProcessLoop( &mqttContext, 0 );
+
+                    pthread_mutex_unlock( &mqttMutex );
+                }
+                else
+                {
+                    LogError( ( "Failed to acquire mutex to execute process loop"
+                                ",errno=%s",
+                                strerror( errno ) ) );
+                }
+
+                if( mqttStatus == MQTTSuccess )
+                {
+                    /* Get OTA statistics for currently executing job. */
+                    OTA_GetStatistics( &otaStatistics );
+
+                    LogInfo( ( " Received: %u   Queued: %u   Processed: %u   Dropped: %u",
+                               otaStatistics.otaPacketsReceived,
+                               otaStatistics.otaPacketsQueued,
+                               otaStatistics.otaPacketsProcessed,
+                               otaStatistics.otaPacketsDropped ) );
+                }
+                else
+                {
+                    LogError( ( "MQTT_ProcessLoop returned with status = %s.",
+                                MQTT_Status_strerror( mqttStatus ) ) );
+
+                    /* Disconnect from broker and close connection. */
+                    disconnect();
+
+                    /* Set connection flag to false. */
+                    mqttSessionEstablished = false;
+
+                    /* Suspend OTA operations. */
+                    otaRet = OTA_Suspend();
+
+                    if( otaRet != OtaErrNone )
```

Comment:
It looks like the rest of the blocks in this function follow the pattern:
```
if( success )

else 
    //handle error
```
but this one handles the error first. Would it be better if it were consistent?

Reply:
Addressed in https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1418/commits/5c73f4f2a45cd02b6240f46466f7473ed8ddb151